File: src/com/leetcode/Medium/MinimumNumberOfSwapsToMakeTheStringBalanced.java
package com.leetcode.Medium;

// 1963. Minimum Number of Swaps to Make the String Balanced
public class MinimumNumberOfSwapsToMakeTheStringBalanced {
    public int minSwaps(String s) {
        int stackSize = 0;
        for (char ch : s.toCharArray()) {
            if (ch == '[') {
                stackSize++;
            } else {
                if (stackSize > 0)
                    stackSize--;
            }
        }

        return (stackSize + 1) / 2;
    }
}

File: JavaScript/Medium/MinimumNumberOfSwapsToMakeTheStringBalanced.js
/**
 * 1963. Minimum Number of Swaps to Make the String Balanced
 * @param {string} s
 * @return {number}
 */
const minSwaps = function (s) {
  let stackSize = 0;
  for (let ch of s) {
    if (ch === '[') {
      stackSize++;
    } else {
      if (stackSize > 0)
        stackSize--;
    }
  }

  return Math.floor((stackSize + 1) / 2);
};